ghga_connector/core/main.py

```diff
@@ -230,19 +256,22 @@ def download_parts(
     output_file: str,
     part_size: int,
     file_size: int,
-    public_key: str,
 ) -> None:
     """
     Downloads a file using a specific download_url to download all its parts.
     """
 
     download_urls = get_download_urls(
-        api_url=api_url, file_id=file_id, public_key=public_key
+        api_url=api_url,
+        file_id=file_id,
     )
     file_parts = download_file_parts(
         download_urls=download_urls, part_size=part_size, total_file_size=file_size
     )
 
     with open(output_file, "wb") as file:
```

Comment:
This needs to be opened in append mode, else we're truncating and missing the envelope

Reply:
Ha. Realized this miself in the above comment :-D